zio-http/shared/src/main/scala/zio/http/ErrorResponseConfig.scala

```diff
@@ -0,0 +1,43 @@
+package zio.http
+
+import zio.{FiberRef, Unsafe}
+
+/**
+ * Configuration for the response generation
+ *
+ * @param withErrorBody
+ *   if true, includes the error message in the response body
+ * @param withStackTrace
+ *   if true, includes the stack trace in the response body
+ * @param maxStackTraceDepth
+ *   maximum number of stack trace lines to include in the response body. Set to
+ *   0 to include all lines.
+ */
+case class ErrorResponseConfig(
+  withErrorBody: Boolean = false,
+  withStackTrace: Boolean = false,
+  maxStackTraceDepth: Int = 10,
+  errorFormat: ErrorResponseConfig.ErrorFormat = ErrorResponseConfig.ErrorFormat.Html,
```

Comment:
I'm thinking plaintext might make more sense as a default as it accounts for a wider range of usecases (APIs & serving user content)